api/api.go

```diff
@@ -0,0 +1,321 @@
+        log.Printf("Error happened while responding to GetArticlesHistory: %v", err)
+    }
+}
+
+func (a *UnarXivApi) GetSearchHistory(w http.ResponseWriter, r *http.Request) {
+    authData := usecases.DummyAuthenticationData // TODO extract auth from headers
+    response, err := a.Core.GetSearchHistory(&authData)
+    if err != nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        log.Printf("Error happened in Core.GetSearchHistory: %v", err)
+        return
+    }
+
+    if err := RespondWithJSON(w, &response, http.StatusOK); err != nil {
+        log.Printf("Error happened while responding to GetSearchHistory: %v", err)
+    }
+}
+
+func (a *UnarXivApi) GetSearchQueriesUpdates(w http.ResponseWriter, r *http.Request) {
+    authData := usecases.DummyAuthenticationData // TODO extract auth from headers
+    response, err := a.Core.GetSearchQueriesUpdates(&authData)
+    if err != nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        log.Printf("Error happened in Core.GetSearchQueriesUpdates: %v", err)
+        return
+    }
+
+    if err := RespondWithJSON(w, &response, http.StatusOK); err != nil {
+        log.Printf("Error happened while responding to GetSearchQueriesUpdates: %v", err)
+    }
+}
+
+func (a *UnarXivApi) GetArticlesUpdates(w http.ResponseWriter, r *http.Request) {
+    authData := usecases.DummyAuthenticationData // TODO extract auth from headers
+    response, err := a.Core.GetArticlesUpdates(&authData)
+    if err != nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        log.Printf("Error happened in Core.GetArticlesUpdates: %v", err)
+        return
+    }
+
+    if err := RespondWithJSON(w, &response, http.StatusOK); err != nil {
+        log.Printf("Error happened while responding to GetArticlesUpdates: %v", err)
+    }
+}
+
+func (a *UnarXivApi) GetArticleSubscriptionStatus(w http.ResponseWriter, r *http.Request) {
+    var getArticleSubscriptionStatusRequest usecases.GetArticleSubscriptionStatusRequest
+    if err := r.ParseForm(); err != nil {
+        w.WriteHeader(http.StatusBadRequest)
+        log.Printf("Error happened while parsing form params: %v", err)
+        return
+    }
+    getArticleSubscriptionStatusRequest.ArticleId = r.Form.Get("articleId")
+    getArticleSubscriptionStatusRequest.AuthData = usecases.DummyAuthenticationData // TODO extract auth from headers
+
+    response, err := a.Core.GetArticleSubscriptionStatus(&getArticleSubscriptionStatusRequest)
+    if err != nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        log.Printf("Error happened in Core.GetArticleSubscriptionStatus: %v", err)
+        return
+    }
+
+    if err := RespondWithJSON(w, &response, http.StatusOK); err != nil {
+        log.Printf("Error happened while responding to GetArticlesSubscriptionStatus: %v", err)
+    }
+}
+
+func (a *UnarXivApi) PostArticleSubscriptionStatus(w http.ResponseWriter, r *http.Request) {
```

Comment:
Could you explain the meaning of this method?
Does this allow to subscribe and unsubscribe?

Reply:
Yes. Depending on the flag.

Reply:
Yes, this method sets user's subscription status, i.e. whether they are subscribed or not.

Reply:
I know it sounds weird, but if we talking about REST-ish API we should try to express this logic by manipulating resources. 

E.g.: we can create a new resource with subscription. POST /??/subscriptions  
We get a new resource with id. And then we can DELETE /??/subscriptions/<id>

Reply:
fixed!